### ISSUE TITLE
X-Forwarded-For, log sanitization and header parsing

### ISSUE THREAD

**Issue by dualbus** [maxburke/oneshot#5]
With the introduction of `X-Forwarded-For` logging, it also opened up for clients to forge this header and inject arbitrary garbage into the output log.

For example, the sample nginx configuration adds:

    proxy_set_header    X-Forwarded-For $proxy_add_x_forwarded_for;

Which according to http://nginx.org/en/docs/http/ngx_http_proxy_module.html#variables

    the “X-Forwarded-For” client request header field with the $remote_addr variable appended to it, separated by a comma. [...]

Which means that a client can send a forged X-Forwarded-For with garbage, then nginx will add its own address, and oneshot will end up logging the garbage.

Also, header parsing seems to be incomplete. HTTP allows for line continuations, i.e. https://tools.ietf.org/html/rfc2616#section-4.2

    Each header field consists of a name followed by a colon (":") and the field
    value. Field names are case-insensitive. The field value MAY be preceded by any
    amount of LWS, though a single SP is preferred. Header fields can be extended
    over multiple lines by preceding each extra line with at least one SP or HT.

**Comment by maxburke** [comment]
Improved X-Forwarded-For handling: 12303b4

```
max@homer:~/src/oneshot/src$ curl -H "X-Forwarded-For: gandalf, 10.10.32.4" http://localhost:8080
```

```
max@homer:~/src/oneshot$ ./oneshot
oneshot[24145]: 10.10.32.4 - [2017-03-11:12:56:17] "GET / HTTP/1.1" 301 98
```

**Comment by maxburke** [comment]
I've broken the header parsing shortfall into a separate issue.

**Comment by maxburke** [comment]
... which should now be fixed as of e6016ee ...
